substrate/frame/assets-vesting/src/lib.rs

```diff
@@ -0,0 +1,786 @@
+				.max(T::WeightInfo::vest_unlocked(T::MAX_VESTING_SCHEDULES))
+		)]
+		pub fn vest(origin: OriginFor<T>, asset: AssetIdOf<T, I>) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			Self::do_vest(asset, who)
+		}
+
+		/// Unlock any vested funds of a `target` account.
+		///
+		/// The dispatch origin for this call must be _Signed_.
+		///
+		/// - `asset`: Id of the asset class of the asset account for which the vesting applies.
+		/// - `target`: The account whose vested funds should be unlocked. Must have funds still
+		/// frozen under this pallet.
+		///
+		/// Emits either `VestingCompleted` or `VestingUpdated`.
+		///
+		/// ## Complexity
+		/// - `O(1)`.
+		#[pallet::call_index(1)]
+		#[pallet::weight(T::WeightInfo::vest_other_locked(T::MAX_VESTING_SCHEDULES)
+				.max(T::WeightInfo::vest_other_unlocked(T::MAX_VESTING_SCHEDULES))
+			)]
+		pub fn vest_other(
+			origin: OriginFor<T>,
+			asset: AssetIdOf<T, I>,
+			target: AccountIdLookupOf<T>,
+		) -> DispatchResult {
+			ensure_signed(origin)?;
+			let who = T::Lookup::lookup(target)?;
+			Self::do_vest(asset, who)
+		}
+
+		/// Create a vested transfer.
+		///
+		/// The dispatch origin for this call must be _Signed_.
+		///
+		/// - `asset`: Id of the asset class of the asset account for which the vesting applies.
+		/// - `target`: The account receiving the vested funds.
+		/// - `schedule`: The vesting schedule attached to the transfer.
+		///
+		/// Emits `VestingCreated`.
+		///
+		/// NOTE: This will unlock all schedules through the current block.
+		///
+		/// ## Complexity
+		/// - `O(1)`.
+		#[pallet::call_index(2)]
+		#[pallet::weight(T::WeightInfo::vested_transfer(T::MAX_VESTING_SCHEDULES))]
+		pub fn vested_transfer(
+			origin: OriginFor<T>,
+			asset: AssetIdOf<T, I>,
+			target: AccountIdLookupOf<T>,
+			schedule: VestingInfo<BalanceOf<T, I>, BlockNumberFor<T>>,
+		) -> DispatchResult {
+			let transactor = ensure_signed(origin)?;
+			let target = T::Lookup::lookup(target)?;
+			Self::do_vested_transfer(asset, &transactor, &target, schedule)
+		}
+
+		/// Force a vested transfer.
+		///
+		/// The dispatch origin for this call must be `ForceOrigin`.
+		///
+		/// - `asset`: Id of the asset class of the asset account for which the vesting applies.
+		/// - `source`: The account whose funds should be transferred.
+		/// - `target`: The account that should be transferred the vested funds.
+		/// - `schedule`: The vesting schedule attached to the transfer.
+		///
+		/// Emits `VestingCreated`.
+		///
+		/// NOTE: This will unlock all schedules through the current block.
+		///
+		/// ## Complexity
+		/// - `O(1)`.
+		#[pallet::call_index(3)]
+		#[pallet::weight(T::WeightInfo::force_vested_transfer(T::MAX_VESTING_SCHEDULES))]
+		pub fn force_vested_transfer(
+			origin: OriginFor<T>,
+			asset: AssetIdOf<T, I>,
+			source: AccountIdLookupOf<T>,
+			target: AccountIdLookupOf<T>,
+			schedule: VestingInfo<BalanceOf<T, I>, BlockNumberFor<T>>,
```

Comment:
we can also add the preservation in argument here, though it is less important.

Reply:
Guess it's not the best there, since these kind of arguments tend to be difficult to understand to extrinsic callers